specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc

```diff
@@ -1000,91 +1000,48 @@ _https://jakarta.ee/specifications/xml-web-services/_ .
-This is typically done by using the _-poa_ option to the _rmic_ command.
-A Jakarta EE product must provide support for these portable _Stub_ and
-_Tie_ classes, typically using the required CORBA POA. However, for
-portability to systems that do not use a POA to implement RMI-IIOP,
-applications should not depend on the fact that the _Tie_ extends the
-_Servant_ class. A Jakarta EE application that defines or uses RMI-IIOP
-objects other than enterprise beans must include such portable _Stub_
-and _Tie_ classes in the application package. _Stub_ and _Tie_ objects
-for enterprise beans, however, must not be included with the
-application: they will be generated, if needed, by the Jakarta EE product
-at deployment time or at run time.
-
-RMI-IIOP is defined by chapters 5, 6, 13, 15,
-and section 10.6.2 of the CORBA 2.3.1 specification, available at
-_http://www.omg.org/cgi-bin/doc?formal/99-10-07_ , and by the Java™
-Language To IDL Mapping Specification, available at
-_http://www.omg.org/cgi-bin/doc?ptc/2000-01-06_ .
+===== RMI-IIOP
+
+The RMI-IIOP subsystem is composed of APIs that allow for the
+use of RMI-style programming that is independent of the underlying
+protocol.  Implementations these APIs may support the Java SE native RMI
+protocol (JRMP), the CORBA IIOP protocol or any custom protocol that is
+compatible with the RMI programming restrictions (see the RMI-IIOP spec
+for details).
+
+NOTE: The requirements in this section only apply to Jakarta EE products that
+include an Enterprise Beans container with support for remote interfaces.
+
+Jakarta EE applications are required to use the RMI-IIOP APIs (specifically the
+narrow method of javax.rmi.PortableRemoteObject) when accessing
+remote Enterprise JavaBeans components, as described in the Jakarta Enterprise
+Beans 4.0 specification.  This allows Enterprise Beans and their clients to be
+protocol independent and portable to Jakarta EE implementations that may use
+CORBA or RMI as their underlying protocol.  However, there is currently no
+requirement that Enterprise Beans be accessible using the IIOP protocol itself
+or that Jakarta EE implementations support any form of distributed interoperability
+with other Jakarta EE implementations.
+
+The use of the term RMI-IIOP has existed since J2EE 1.2 and EJB 1.1, predates
+any requirements to support CORBA itself and was used to detail application
+programming restrictions necessary to allow application portability.  Requirements
+for implementations to use CORBA IIOP itself were added to J2EE 1.3 and EJB 2.0 and
+expanded the term RMI-IIOP beyond application portability and into distributed
+interoperability. These distributed interoperability requirements
+were marked Proposed Optional in Java EE 1.8 .  As of Jakarta EE 9 and Enterprise
+Beans 4.0 distributed interoperability requirements are fully removed and the term
+RMI-IIOP reverts back to a term that guarantees only application portability, not
+distributed interoperability.
```

Comment:
Overall, I really like this paragraph.  This helps with the history and understanding of how we got where we are.  But, I'm still confused on the app portability vs distributed interop definitions and how they apply to application servers implementing Jakarta EE 9, and how they apply to applications expected to run on Jakarta EE 9 runtimes.

Reply:
Could we end the sentence after `removed`?  

So, something like:
`As of Jakarta EE 9 and Enterprise Beans 4.0 distributed interoperability requirements are fully removed.'

Reply:
As long as we clearly differentiate between distributed interop and application portability (hopefully earlier in this section), then we could end the sentence after "removed".